cmd/centre/main.go

```diff
@@ -69,35 +70,49 @@ type dserver4 struct {
 	hostFile     string
 }
 
-// lookupIP looks up an IP address corresponding to the given address
-// for mac address, prefix a "u", e.g. "u00:11:22:33:44:55"
-func lookupIP(hostFile string, addr string) ([]net.IP, error) {
+// lookupIP looks up an IP address corresponding to the given name.
+// It also returns a slice of other hostnames it found for the same IP.
+func lookupIP(hostFile string, addr string) (net.IP, []string, error) {
 	var err error
 	// First try the override
 	if hostFile != `` {
 		// Read the file and walk each line looking for a match.
 		// We do this so you can update the file without restarting the server
 		var f *os.File
 		if f, err = os.Open(hostFile); err != nil {
-			return nil, err
+			return nil, nil, err
 		}
+		defer f.Close()
 		// We're going to be real simple-minded. We take each line to consist of
-		// one IP, followed by one or more whitespace-separated hostnames
+		// one IP, followed by one or more whitespace-separated hostnames, with
+		// the mac address at the end:
+		// <ip> <hostname>... <mac>
 		scan := bufio.NewScanner(f)
 		for scan.Scan() {
 			fields := strings.Fields(scan.Text())
-			if len(fields) < 2 {
+			if len(fields) < 2 || strings.HasPrefix(fields[0], "#") {
 				continue
 			}
-			for _, fld := range fields {
+			var hostnames []string
+			for _, fld := range fields[1:] {
 				if strings.ToLower(fld) == strings.ToLower(addr) {
-					return []net.IP{net.ParseIP(fields[0])}, nil
+					return net.ParseIP(fields[0]), hostnames, nil
```

Comment:
I see we may not be returning all the hostnames, but that is probably fine since we're just using the first one.